python/cuml/dask/naive_bayes/naive_bayes.py

```diff
@@ -219,68 +217,22 @@ def _get_part(parts, idx):
     def _get_size(arrs):
         return arrs.shape[0]
 
-    def predict(self, X):
-
+    def predict(self, X, delayed=True, parallelism=5):
```

Comment:
What if we only allowed delayed=True and required callers to wait for the results? Maybe that would be more typical of Dask APIs?

Reply:
I agree, this does make more sense now that we don't need the eager code path at all.